### PR TITLE
Effective attachment

### PR DESCRIPTION
**Pull request by tsnoam**

### CODE REVIEW

File: telegram/message.py [hunk 1]
```diff
@@ -301,6 +305,31 @@ def de_json(cls, data, bot):
 
         return cls(bot=bot, **data)
 
+    @property
+    def effective_attachment(self):
+        """
+        Optional[Union[:class:`telegram.Audio`, :class:`telegram.Game`, :class:`telegram.Document`,
+         :class:`telegram.Photo`, :class:`telegram.Sticker`, :class:`telegram.Video`,
```
Comment (Eldinnie) [inline telegram/message.py]: telegram.Photo does not exists. Should be List[:class:\`telegram.PhotoSize\`]

File: telegram/message.py [hunk 2]
```diff
@@ -301,6 +305,31 @@ def de_json(cls, data, bot):
...
+         :class:`telegram.Voice`, :class:`telegram.VideoNote`, :class:`telegram.Contact`,
+         :class:`telegram.Location`, :class:`telegram.Venue`, :class:`telegram.Invoice`,
+         :class:`telegram.SuccessfulPayment`]]: The attachment that this message was sent with. May
+          be ``None`` if no attachment was sent.
```
Comment (Eldinnie) [inline telegram/message.py]: space too much at start of line, makes weird docs.

File: telegram/message.py [hunk 3]
```diff
@@ -301,6 +305,31 @@ def de_json(cls, data, bot):
 
         return cls(bot=bot, **data)
 
+    @property
+    def effective_attachment(self):
+        """
+        Optional[Union[:class:`telegram.Audio`, :class:`telegram.Game`, :class:`telegram.Document`,
```
Comment (bomjacob) [inline telegram/message.py]: Shouldn't it be:
```
Returns:
    :class:`telegram.Audio` | :class:`telegram.Game` | :class:`telegram.Document` [...]
```
I'm not sure our sphinx parser understands the optional and union very well

**Comment by Eldinnie** [comment]
I thought I had put this here, but apparently I only mentioned it in chat. photo is of type List[:class:'telegram.PhotoSize'] and I would like it visually better if the possible attachments would be listed in alphabetical order.

**Pull request merged**